ajax/entitytreesons.php

```diff
@@ -52,7 +52,7 @@
 $ancestors = getAncestorsOf('glpi_entities', $_SESSION['glpiactive_entity']);
 
 $ckey    = 'entity_selector';
-$subckey = sha1($_SESSION['glpiactiveentities_string']);
+$subckey = sha1($_SESSION['glpiactiveentities_string'] . json_encode($_SESSION['glpiactiveprofile']['entities']));
```

Comment:
I mauy be wrong, but as data is build from `$_SESSION['glpiactiveprofile']['entities']` and `$_SESSION['glpiactive_entity']`, the cache key should probably be based on these elements.

Also, to enhance caching performances, we could remove `if ($entities_id == $_SESSION['glpiactive_entity']) { $entity['selected'] = 'true'; }` from the tree building function, and apply this attribute on result fetched from cache, so we would be able to use a unique cache entry for each profile, and probably use `$_SESSION['glpiactiveprofile']['id']` as a cache key.

Reply:
Enhancing cache performances is propably out of the scope of this PR

Reply:
In this case, we should, at least, ensure that case key is based on values that may affect result.
```suggestion
$subckey = sha1($_SESSION['glpiactive_entity'] . json_encode($_SESSION['glpiactiveprofile']['entities']));
```

Reply:
Yes, seems it should be better